File: python-solid-principle-sample-code/com/principle/solid/interface_segregation/ISPMain.py
from typing import List

from Shape import Shape
from Cube import Cube
from Rectangle import Rectangle
from ThreeDimensionalShape import ThreeDimensionalShape


def main():
    print('Hello World')

    cube: Cube = Cube(10)
    rectangle: Rectangle = Rectangle(10, 20)

    shapes: List[Shape] = [cube, rectangle]

    for shape in shapes:
        print("{{shapes_sum: {}}}".format(int(shape.area())))

    for shape in shapes:
        if isinstance(shape, ThreeDimensionalShape):
            print("{{shapes_volume: {}}}".format(int(shape.volume())))


if __name__ == "__main__":
    main()

File: java-solid-principle-sample-code/src/com/principle/solid/interface_segregation/ISPMain.java
package com.principle.solid.interface_segregation;

import java.util.Arrays;
import java.util.List;

public class ISPMain {

	public static void main(String[] args) {
		System.out.println("Hello World");
		
		Cube cube = new Cube(10);
		Rectangle rectangle = new Rectangle(10,20);
		
		// Java 8
		List<Shape> shapes = Arrays.asList(cube, rectangle);

		for (Shape shape : shapes) {
			System.out.println(String.format("{shapes_sum: %s}", shape.area()));			
		}

		for (Shape shape : shapes) {
			if(shape instanceof ThreeDimensionalShape) {
				System.out.println(String.format("{shapes_volume: %s}", ((ThreeDimensionalShape)shape).volume()));				
			}
		}

	}
}
